### PR TITLE
Moving conf editor to the shell

### LINKED ISSUE #360: Replace the conf editor with TWR "dotfile" command
**What's missing? Please describe your flow and where this feature will help.**
I don't like the cogwheels button not going through TWR.
I want it to open TWR and lunch vim with the dotfile in TWR.

**Additional context**

It's probably best to use WASM and compile one of the existing editor.
There's an vim effort which https://github.com/rhysd/vim.wasm is not supported
on mobile browsers.

The best bet is to take simple version, like the original [ex](https://github.com/dlisboa/ex) and compule it to wasm. As it runs inside the TWR terminal it needs no rendering,
just a connection to the terminal write method and keyboard event.

### PR DESCRIPTION
**Pull request by eyal-sasson**
Resolves #360. Added `config` command, and moved the rendering of the editor to override t0.
Still need to move all functionality to the shell.

### CODE REVIEW

**Comment by eyal-sasson** [comment]
Functionality is basically done, still thinking how closing the editor should be handled.

File: src/terminal7.js [hunk 1]
```diff
@@ -1068,4 +1030,22 @@ export class Terminal7 {
         this.lastIdVerify = now
         return this.keys
     }
+    async getDotfile() {
+        return (await Preferences.get({key: "dotfile"})).value || DEFAULT_DOTFILE
+    }
+    saveDotfile() {
```
Comment (daonb) [inline src/terminal7.js:1036]: I'm not sure how easy it is, but its much nicer to pass the conf as a parameter and let the function update `this.conf`

**Review by daonb — CHANGES_REQUESTED** [review]

**Comment by daonb** [comment]
> Functionality is basically done, still thinking how closing the editor should be handled.

How about before opening the editor we print a message on TWR: `Opening vi-style editor. Use ":w" to save & exit and ":q" to quit. Hit enter to continue"

(I hope we can add the ":q" command to codemirror)

File: index.html [hunk 2]
```diff
@@ -20,6 +20,10 @@
       <div id="log" class="hidden">
           <div> <div id="t0">
                   <div id="capslock-indicator" class="hidden"><i class="f7-icons">exclamationmark_triangle</i> CAPS</div>
+                  <div id="settings" class="hidden">
+                  <textarea id="edit-conf" lines="20" cols="40">
```
[the remark this answers is not in the document]
Comment (eyal-sasson) [reply index.html:24]: Don't think it matters as the element is hidden when the editor is closed and when it's open it fits automatically.
I'll just remove the rows and cols altogether.

File: src/terminal7.js (hunk 1 above)
Comment (eyal-sasson) [reply src/terminal7.js:1036]: We already have `loadConf` for updating `this.conf`, but maybe we should change `saveDotfile` to `loadAndSaveConf` (though it doesn't fit well imo).

**Comment by eyal-sasson** [comment]
I've looked into it and unfortunately I don't think we can add `:q`, so the only way to discard would be clicking the conf button again.

**Comment by eyal-sasson** [comment]
Just found `CodeMirror.Vim.defineEx()` so I'll add `:q` as well.

**Comment by daonb** [comment]
Looks much better. Two problems I found is that it's still not resizing on narrow screens and that we lost the link to the document with all the conf options. We should probably add the link to the line with ":w" and ":q".

**Pull request merged**